core/coreapi/path.go

```diff
@@ -73,3 +74,8 @@ func (api *CoreAPI) ResolvePath(ctx context.Context, p coreiface.Path) (coreifac
 
 	return coreiface.NewResolvedPath(ipath, node, root, gopath.Join(rest...)), nil
 }
+
+func (api *CoreAPI) getSession(ctx context.Context) *CoreAPI {
```

Comment:
Let's add some documentation here. Specifically, we need to *at least* explain that the returned API has a read-only DAG.